File: algo/169. Majority Element.py
# my solution used too much extra space

class Solution:
    def majorityElement(self, nums: 'List[int]') -> 'int':
        wc = dict()
        for num in nums:
            wc[num] = wc.get(num, 0) + 1
        size = len(nums)
        for k,v in wc.items():
            if v > size//2:
                return k

File: round3/169. Majority Element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int cnt = 1, curr = nums[0];
        
        for (int i = 1; i < nums.Length; i ++) {
            if (nums[i] == curr) {
                cnt ++;
            } else {
                cnt --;
                
                if (cnt == 0) {
                    curr = nums[i + 1];
                }
            }
        }
        
        return curr;
    }
}